Fix Compiler Warning in QubitQIRExecutionManager

```bash
[59/257] Building CXX obj...QIRExecutionManager.cpp.o
/workspaces/cuda-quantum/runtime/cudaq/qis/managers/qir/QubitQIRExecutionManager.cpp:190:28: warning: '(anonymous namespace)::QIRExecutionManager::measure' hides overloaded virtual function [-Woverloaded-virtual]
  cudaq::SpinMeasureResult measure(cudaq::spin_op &op) override {
                           ^
/workspaces/cuda-quantum/runtime/cudaq/qis/managers/BasicExecutionManager.h:253:7: note: hidden overloaded virtual function 'cudaq::BasicExecutionManager::measure' declared here: type mismatch at 1st parameter ('const cudaq::QuditInfo &' vs 'cudaq::spin_op &')
  int measure(const cudaq::QuditInfo &target) override {
      ^
1 warning generated.
```

--- a/.github/workflows/config/spelling_allowlist.txt
+++ b/.github/workflows/config/spelling_allowlist.txt
@@ -87,6 +87,7 @@ discretization
 discretize
 discretized
 eigenstate
+enqueues
 ensmallen
 entangler
 entanglers
@@ -120,6 +121,7 @@ nvq
 pre
 precompute
 precomputed
+prepend
 qir
 qke
 qpp

--- a/runtime/cudaq/qis/managers/BasicExecutionManager.h
+++ b/runtime/cudaq/qis/managers/BasicExecutionManager.h
@@ -18,11 +18,11 @@
 
 namespace cudaq {
 
-/// @brief The BasicExecutionManager provides a common base class for
-/// specializations that implements the ExecutionManager type. Most of the
-/// required ExecutionManager functionality is implemented here, with
+/// @brief The `BasicExecutionManager` provides a common base class for
+/// specializations that implement the `ExecutionManager` type. Most of the
+/// required `ExecutionManager` functionality is implemented here, with
 /// backend-execution-specific details left for further subtypes. This type
-/// enqueues all quantum operations and flush them at specific synchronization
+/// enqueues all quantum operations and flushes them at specific synchronization
 /// points. Subtypes should implement concrete operation execution, qudit
 /// measurement, allocation, and deallocation, and execution context handling
 /// (e.g. sampling)
@@ -35,7 +35,7 @@ protected:
       std::tuple<std::string, std::vector<double>,
                  std::vector<cudaq::QuditInfo>, std::vector<cudaq::QuditInfo>>;
 
-  /// @brief Typedef for a queue of instructions
+  /// @brief `typedef` for a queue of instructions
   using InstructionQueue = std::queue<Instruction>;
 
   /// @brief The current execution context, e.g. sampling
@@ -203,7 +203,7 @@ public:
     // Make a copy of the name that we can mutate if necessary
     std::string mutable_name(gateName);
 
-    // Make a copy of the params that we can mutate
+    // Make a copy of the parameters that we can mutate
     std::vector<double> mutable_params = params;
 
     // Create an array of controls, we will

--- a/runtime/cudaq/qis/managers/BasicExecutionManager.h
+++ b/runtime/cudaq/qis/managers/BasicExecutionManager.h
@@ -81,6 +81,8 @@ protected:
   /// @brief Subtype-specific method for performing qudit measurement.
   virtual int measureQudit(const cudaq::QuditInfo &q) = 0;
 
+  virtual void measureSpinOp(const cudaq::spin_op &op) = 0;
+
 public:
   BasicExecutionManager() = default;
   virtual ~BasicExecutionManager() = default;
@@ -256,6 +258,13 @@ public:
 
     // Instruction executed, run the measure call
     return measureQudit(target);
+  }
+
+  cudaq::SpinMeasureResult measure(cudaq::spin_op &op) override {
+    synchronize();
+    measureSpinOp(op);
+    return std::make_pair(executionContext->expectationValue.value(),
+                          executionContext->result);
   }
 };
 

--- a/runtime/cudaq/qis/managers/qir/QubitQIRExecutionManager.cpp
+++ b/runtime/cudaq/qis/managers/qir/QubitQIRExecutionManager.cpp
@@ -21,7 +21,7 @@
 #include <stack>
 
 namespace {
-Array *spinToArray(cudaq::spin_op &);
+Array *spinToArray(const cudaq::spin_op &);
 
 /// The QIRQubitQISManager will implement allocation, deallocation, and
 /// quantum instruction application via calls to the extern declared QIR
@@ -184,27 +184,22 @@ protected:
     return res ? 1 : 0;
   }
 
+  void measureSpinOp(const cudaq::spin_op &op) override {
+    Array *term_arr = spinToArray(op);
+    __quantum__qis__measure__body(term_arr, nullptr);
+    return;
+  }
+
 public:
   QIRExecutionManager() = default;
   virtual ~QIRExecutionManager() {}
-
-  cudaq::SpinMeasureResult measure(cudaq::spin_op &op) override {
-    synchronize();
-    // FIXME need to remove QIR things from spin_op
-    Array *term_arr = spinToArray(op);
-    __quantum__qis__measure__body(term_arr, nullptr);
-    // auto counts_raw = ctx->extract_results();
-    auto exp = executionContext->expectationValue;
-    auto data = executionContext->result;
-    return std::make_pair(exp.value(), data);
-  }
 
   void resetQudit(const cudaq::QuditInfo &id) override {
     __quantum__qis__reset(qubits[id.id]);
   }
 };
 
-Array *spinToArray(cudaq::spin_op &op) {
+Array *spinToArray(const cudaq::spin_op &op) {
   // How to pack the data???
   // add all term data as correct pointer to double for x,y,z,or I.
   // After each term add a pointer to real part of term coeff,

--- a/runtime/cudaq/qis/managers/qir/QubitQIRExecutionManager.cpp
+++ b/runtime/cudaq/qis/managers/qir/QubitQIRExecutionManager.cpp
@@ -187,7 +187,6 @@ protected:
   void measureSpinOp(const cudaq::spin_op &op) override {
     Array *term_arr = spinToArray(op);
     __quantum__qis__measure__body(term_arr, nullptr);
-    return;
   }
 
 public:

--- a/runtime/cudaq/spin/matrix.cpp
+++ b/runtime/cudaq/spin/matrix.cpp
@@ -17,7 +17,7 @@
 namespace cudaq {
 
 /// @brief Hash function for an Eigen::MatrixXcd
-struct complex_matrix_hash : std::unary_function<Eigen::MatrixXcd, size_t> {
+struct complex_matrix_hash {
   std::size_t operator()(const Eigen::MatrixXcd &matrix) const {
     size_t seed = 0;
     for (Eigen::Index i = 0; i < matrix.size(); ++i) {

--- a/unittests/qudit/simple_qudit/SimpleQuditExecutionManager.cpp
+++ b/unittests/qudit/simple_qudit/SimpleQuditExecutionManager.cpp
@@ -92,7 +92,7 @@ protected:
     return measurement_result;
   }
 
-  void measureSpinOp(const cudaq::spin_op &op) override { return; }
+  void measureSpinOp(const cudaq::spin_op &) override {}
 
 public:
   SimpleQuditExecutionManager() {

--- a/unittests/qudit/simple_qudit/SimpleQuditExecutionManager.cpp
+++ b/unittests/qudit/simple_qudit/SimpleQuditExecutionManager.cpp
@@ -92,6 +92,8 @@ protected:
     return measurement_result;
   }
 
+  void measureSpinOp(const cudaq::spin_op &op) override { return; }
+
 public:
   SimpleQuditExecutionManager() {
     instructions.emplace("plusGate", [&](const Instruction &inst) {

